File: packages/ios/Sources/SeasonSprint/GameMode.swift
import Foundation

/// Which game mode a records/season view is scoped to. Mirrors the server's `GameMode` type.
enum GameMode: String, Sendable, CaseIterable {
    case worldTour = "world-tour"
    case ranked = "ranked"

    var wireValue: String { rawValue }

    var prefsSuffix: String {
        switch self {
        case .worldTour: return "worldTour"
        case .ranked: return "ranked"
        }
    }

    var label: String {
        switch self {
        case .worldTour: return "World Tour"
        case .ranked: return "Ranked"
        }
    }
}

File: packages/android/app/src/main/java/com/lcarthur/seasonsprint/GameMode.kt
package com.lcarthur.seasonsprint

/** Which game mode a records/season view is scoped to. Mirrors the server's `GameMode` type. */
enum class GameMode(val wireValue: String, val prefsSuffix: String, val label: String) {
    WorldTour("world-tour", "worldTour", "World Tour"),
    Ranked("ranked", "ranked", "Ranked"),
}
